### PR TITLE
Kill the remnants in the name of math

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Describe the solution
Further porting of old syntax into math, this time "u_compare_var", "npc_compare_var", "u_adjust_var", "npc_adjust_var" was ported
#### Testing
Spawned in a world, tested a titbit

### CODE REVIEW

File: data/json/npcs/TALK_TEST.json [hunk 1]
```diff
@@ -682,84 +682,84 @@
     "responses": [
       { "text": "This is a basic test response.", "topic": "TALK_DONE" },
       {
-        "text": "This is a u_adjust_var test response that increments by 1.",
+        "text": "This is a math test response that increments by 1.",
```
Comment (andrei8l) [inline data/json/npcs/TALK_TEST.json:685]: Better remove this object (`TALK_TEST_ADJUST_VARS`) and its associated test (`npc_talk_adjust_vars`).

**Comment by John-Candlebury** [comment]
Hmm could you please playtest the Aftershock motion sensor with these changes? I'm not fully certain it'll work with the new logic.

The proper behavior is: If you turn it on while its wielded and your character  is surroundedwith hostile monsters that you cant see. the monsters should periodically show as red '?' on the map for a short while every 5 seconds or so,

**Comment by andrei8l** [comment]
> The proper behavior is: If you turn it on while its wielded and your character is surroundedwith hostile monsters that you cant see. the monsters should periodically show as red '?' on the map for a short while every 5 seconds or so,

Isn't this working correctly?

[output.webm](https://github.com/CleverRaven/Cataclysm-DDA/assets/68240139/65c51e05-c4c2-43fd-b0bc-879c4526f129)

Change this to math too, this the motion sensor came up:
https://github.com/CleverRaven/Cataclysm-DDA/blob/a80490e2797b832f0777138ba4235c7fe5fa1c98/data/mods/Aftershock/effects_on_condition.json#L48

**Comment by andrei8l** [comment]
However, I see that `u_adjust_var` and `u_set_var` trigger the event bus and math doesn't do that. I'll have to fix it before this PR can go

File: data/json/npcs/TALK_TEST.json [hunk 2]
```diff
@@ -961,14 +873,14 @@
     "dynamic_line": "This is a test conversation that shouldn't appear in the game.",
     "responses": [
       {
-        "text": "This is a u_adjust_var test response that increments by 1.",
+        "text": "This is a math test response that increments by 1.",
         "topic": "TALK_DONE",
-        "effect": { "u_adjust_var": "test", "type": "test", "context": "var_compare_test", "adjustment": 1 }
+        "effect": { "math": [ "u_test_var_adjust_test_test", "++" ] }
```
Comment (andrei8l) [inline data/json/npcs/TALK_TEST.json:878]:
```suggestion
        "effect": { "math": [ "u_test_var_compare_test_test", "++" ] }
```

File: data/json/npcs/TALK_TEST.json [hunk 3]
```diff
@@ -961,14 +873,14 @@
...
       },
       {
-        "text": "This is an npc_adjust_var test response that increments by 2.",
+        "text": "This is an math test response that increments by 2.",
         "topic": "TALK_DONE",
-        "effect": { "npc_adjust_var": "test", "type": "test", "context": "var_compare_test", "adjustment": 2 }
+        "effect": { "math": [ "u_test_var_adjust_test_test", "+=", "2" ] }
```
Comment (andrei8l) [inline data/json/npcs/TALK_TEST.json:883]:
```suggestion
        "effect": { "math": [ "n_test_var_compare_test_test", "+=", "2" ] }
```

**Review by LyleSY — APPROVED** [review]

**Pull request merged**